sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -48,6 +54,9 @@ export class CASSupplierIntegrationService {
     parentProcessSummary: ProcessSummary,
     studentSuppliers: StudentSupplierToProcess[],
   ): Promise<number> {
+    // Force the CAS token to be acquired before starting the process.
+    await this.casService.getToken();
```

Comment:
Minor and not a blocker, the method we are looking here is like `initiToken` right?

Reply:
This is the same method used internally by the method to get a token (request one or cached one).
There is no guarantee that this method will be the one initializing the token, for instance, if the scheduler is retrying due to some individual student error the token will still be cached.

The method getToken can be kept private (as it was before) and we can create something like `ensureTokenInitialization`.
I reused the same method for simplicity. Please let me know if it would be a blocker or if it requires a quick chat 😉